src/execution/compiler/operator/index_join_translator.cpp

```diff
@@ -87,13 +98,27 @@ void IndexJoinTranslator::PerformPipelineWork(WorkContext *context, FunctionBuil
       // PARENT_CODE
       context->Push(function);
     }
+
+    CounterAdd(function, num_scans_index_, 1);
   }
   loop.EndLoop();
 
+  CounterSetExpr(function, index_size_,
+                 GetCodeGen()->CallBuiltin(ast::Builtin::IndexIteratorGetSize, {GetCodeGen()->AddressOf(index_iter_)}));
   // @indexIteratorFree(&index_iter_)
   FreeIterator(function);
 }
 
+void IndexJoinTranslator::FinishPipelineWork(const Pipeline &pipeline, FunctionBuilder *function) const {
+  FeatureRecord(function, brain::ExecutionOperatingUnitType::IDX_SCAN,
+                brain::ExecutionOperatingUnitFeatureAttribute::NUM_ROWS, pipeline, CounterVal(index_size_));
+  FeatureRecord(function, brain::ExecutionOperatingUnitType::IDX_SCAN,
+                brain::ExecutionOperatingUnitFeatureAttribute::CARDINALITY, pipeline, CounterVal(num_scans_index_));
```

Comment:
To match the models, can we change this to per-loop num scans, if num loops > 0 (i.e., int(num_scans_index_ / num_loops_))?